### ISSUE TITLE
Tabs block: Accessibility: tabpanels use of tabindex

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#75403]
### Description

When a tabpanel is active, it's given `tabindex="-1"`. Active panels should generally have `tabindex="0"` or no tabindex attribute, depending on whether the starting content inside the panel is focusable. 

It should be feasible to test the contents to identify whether the first contained element is focusable, but if not, it would be better to give active panels `tabindex="0"`, so that focus can move directly to them.

Inactive tabpanels currently have `tabindex="0"` and the `hidden` attribute. This just doesn't really make any sense; hidden objects shouldn't be focusable. Most devices should give preference to the hidden state and ignore the tabindex, but this isn't something that can be guaranteed. Inactive tabpanels should not have `tabindex="0"`.

### Step-by-step reproduction instructions

1. Create a tab block with at least two tabs
2. Activate one tab
3. View the source and observe the tabindex & active states.

### Screenshots, screen recording, code snippet

<img width="901" height="359" alt="Image" src="https://github.com/user-attachments/assets/631fbc6f-90be-4be4-bde6-a3e5c0d221f8" />

### Environment info

-WP 7.0-alpha-61605
-Gutenberg 22.5.1
-MacOS/Firefox 146.0.1

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by joedolson** [comment]
@t-hamano I'd consider this a blocking issues for 7.0.

**Comment by t-hamano** [comment]
> It should be feasible to test the contents to identify whether the first contained element is focusable

This should be possible. For example, the Navigation Block has a process that extracts focusable elements in a menu and finds the first or last focusable element.

https://github.com/WordPress/gutenberg/blob/bdb94463eeac7d07cbbe8879084ca16d3b256dbd/packages/block-library/src/navigation/view.js#L223-L227

Similarly, it shouldn't be difficult to determine whether there is a focusable element in a tab panel.

By the way, this can be addressed with just a front-end JS fix, so it won't cause any breaking changes. It would be fine to address this after the Tab Block is stabilized or during the 7.0 beta phase.

**Comment by im3dabasia** [comment]
Thank you to everyone who worked on this issue! I believe it has been addressed by this commit:
https://github.com/WordPress/gutenberg/commit/56ed8258ddc526b894f0f9e9388b0947878c3209 in this PR: https://github.com/WordPress/gutenberg/pull/75484

I think this issue can now be closed, as the tabindex correctly shows 0 when it is in the active state and -1 when it is inactive.
